tests/spaces/test_utils.py

```diff
@@ -135,3 +135,15 @@ def test_flatten_roundtripping(space):
 
     for original, roundtripped in zip(samples, unflattened_samples):
         assert data_equivalence(original, roundtripped)
+
+
+def test_unflatten_discrete_error():
+    value = np.array([0])
+    with pytest.raises(ValueError):
+        utils._unflatten_discrete(gym.spaces.Discrete(1), value)
```

Comment:
Use `utils.unflatten` this will use the correct function so you don't need to specify `_unflatten_discrete`

Reply:
I did this on purpose so that I don't test method dispatch. But no strong opinion, please confirm if you want me to change.

Reply:
Yes, please change

Reply:
Done